src/components/GameItem.js

```diff
@@ -1,29 +1,52 @@
-import React, { Component } from 'react';
-import '../App.css';
-import ItemIcons from '../ItemIcons.js';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../App.css";
+import ItemIcons from "../ItemIcons.js";
+import PropTypes from "prop-types";
 
 class GameItem extends Component {
   static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
-  }
+    types: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
 
+    this.state = {
+      click: false
+    };
+  }
   onItemClicked = () => {
     // Fill this in for Wave 2!
-  }
-    
+    if (this.props.type === "litter") {
+      this.setState({ click: true });
+      this.props.onItemClicked("litter");
```

Comment:
It'd be good to name this something different like onItemClickedCallback to make it easier to differentiate between the 2 functions.